git-discover/src/upwards.rs

```diff
@@ -147,15 +188,19 @@ pub(crate) mod function {
         }
     }
 
-    fn comp_len(c: std::path::Component<'_>) -> usize {
-        use std::path::Component::*;
-        match c {
-            Prefix(p) => p.as_os_str().len(),
-            CurDir => 1,
-            ParentDir => 2,
-            Normal(p) => p.len(),
-            RootDir => 1,
-        }
+    /// Find the number of components parenting the `base_path` before the first directory in `ceiling_dirs`.
+    /// `base_path` needs to be an absolute path. Non-absolute `ceiling_dirs` are discarded if `base_path` is absolute.
+    // TODO: Handle this in a verbatim-path-prefix-neutral way on Windows (introduced by `path::canonicalize`).
```

Comment:
Before I forget: could there be a test that triggers this by failure? It could be marked with `should_panic` or it could be ignored to indicate some work needs to be done to make it pass.

Reply:
Done.